File: app/src/main/java/com/halfminute/itmadness/Words.java
package com.halfminute.itmadness;

import androidx.annotation.NonNull;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class Words {
    @SerializedName("easyWords")
    @Expose
    private final List<GameEasy> easyWords = new ArrayList<>();
    @SerializedName("mediumWords")
    @Expose
    private final List<GameMedium> mediumWords = new ArrayList<>();
    @SerializedName("HardWords")
    @Expose
    private final List<GameHard> hardWords = new ArrayList<>();

    public List<GameEasy> getEasyWords() {
        return easyWords;
    }

    public List<GameMedium> getMediumWords() {
        return mediumWords;
    }

    public List<GameHard> getHardWords() {
        return hardWords;
    }

    @NonNull
    @Override
    public String toString() {
        return "Words{" +
                "easyWords=" + easyWords +
                ", mediumWords=" + mediumWords +
                ", hardWords=" + hardWords +
                '}';
    }
}

record GameEasy(@SerializedName("frWord") @Expose String frWord,
                @SerializedName("nlWord") @Expose String nlWord,
                @SerializedName("enWord") @Expose String enWord) {


    @Override
    public String frWord() {
        return frWord;
    }

    @Override
    public String nlWord() {
        return nlWord;
    }

    @Override
    public String enWord() {
        return enWord;
    }

    @NonNull
    @Override
    public String toString() {
        return "GameEasy{" +
                "frWord='" + frWord + '\'' +
                ", nlWord='" + nlWord + '\'' +
                ", enWord='" + enWord + '\'' +
                '}';
    }
}

record GameMedium(@SerializedName("frWord") @Expose String frWord,
                  @SerializedName("nlWord") @Expose String nlWord,
                  @SerializedName("enWord") @Expose String enWord) {


    @Override
    public String frWord() {
        return frWord;
    }

    @Override
    public String nlWord() {
        return nlWord;
    }

    @Override
    public String enWord() {
        return enWord;
    }

    @NonNull
    @Override
    public String toString() {
        return "GameMedium{" +
                "frWord='" + frWord + '\'' +
                ", nlWord='" + nlWord + '\'' +
                ", enWord='" + enWord + '\'' +
                '}';
    }
}

record GameHard(@SerializedName("frWord") @Expose String frWord,
                @SerializedName("nlWord") @Expose String nlWord,
                @SerializedName("enWord") @Expose String enWord) {


    @Override
    public String frWord() {
        return frWord;
    }

    @Override
    public String nlWord() {
        return nlWord;
    }

    @Override
    public String enWord() {
        return enWord;
    }

    @NonNull
    @Override
    public String toString() {
        return "GameHard{" +
                "frWord='" + frWord + '\'' +
                ", nlWord='" + nlWord + '\'' +
                ", enWord='" + enWord + '\'' +
                '}';
    }
}

File: app/src/main/kotlin/com/halfminute/itmadness/Words.kt
import android.content.Context
import com.google.gson.Gson
import com.google.gson.annotations.SerializedName
import com.halfminute.itmadness.R

enum class Difficulty {
    EASY, MEDIUM, HARD
}

data class Word(
    @SerializedName("frWord") val frWord: String,
    @SerializedName("nlWord") val nlWord: String,
    @SerializedName("enWord") val enWord: String,
    @SerializedName("deWord") val deWord: String
)

data class Words(
    @SerializedName("easyWords") val easyWords: List<Word>,
    @SerializedName("mediumWords") val mediumWords: List<Word>,
    @SerializedName("hardWords") val hardWords: List<Word>
) {
    fun getWordsByDifficulty(difficulty: Difficulty): List<Word> {
        return when (difficulty) {
            Difficulty.EASY -> easyWords
            Difficulty.MEDIUM -> mediumWords
            Difficulty.HARD -> hardWords
        }
    }

    companion object {
        fun getInstance(context: Context): Words? {
            return try {
                val inputStream = context.resources.openRawResource(R.raw.wordjsondata)
                val jsonContent = inputStream.bufferedReader().use { it.readText() }
                Gson().fromJson(jsonContent, Words::class.java)
            } catch (e: Exception) {
                e.printStackTrace()
                null
            }
        }
    }
}

data class LanguageWord(
    @SerializedName("frWord") val frWord: String,
    @SerializedName("nlWord") val nlWord: String,
    @SerializedName("enWord") val enWord: String,
    @SerializedName("deWord") val deWord: String
)

data class LanguageWords(
    @SerializedName("easyWords") val easyWords: List<LanguageWord>,
    @SerializedName("mediumWords") val mediumWords: List<LanguageWord>,
    @SerializedName("hardWords") val hardWords: List<LanguageWord>
) {
    fun getWordsByDifficulty(difficulty: Difficulty): List<LanguageWord> {
        return when (difficulty) {
            Difficulty.EASY -> easyWords
            Difficulty.MEDIUM -> mediumWords
            Difficulty.HARD -> hardWords
        }
    }

    companion object {
        fun getInstance(context: Context): LanguageWords? {
            return try {
                val inputStream = context.resources.openRawResource(R.raw.language_wordjsondata)
                val jsonContent = inputStream.bufferedReader().use { it.readText() }
                Gson().fromJson(jsonContent, LanguageWords::class.java)
            } catch (e: Exception) {
                e.printStackTrace()
                null
            }
        }
    }
}
